### ISSUE TITLE
GPS inaccuracy in recent (3.101 and a few earlier) builds

### ISSUE THREAD

**Issue by jfburke3** [mapillary/mapillary_issues#2719]
I know this isn't a proper bug report, but I think I need to describe it differently than usual.

In the past few releases, including the current 3.101-BETA version, it appears to me that the Mapillary Android app is affecting GPS localization negatively.

As best I can determine, when the Mapillary app is running, my GPS accuracy is almost half what it is when the app isn't active.  My determination of this is other GPS apps I have, including GPS Status and OSM Tracker.  These apps usually--but not always--reflect a GPS accuracy at least a few meters better than Mapillary does.

If I kill Mapillary, then these apps generally show a GPS accuracy in the 1 digit range, even when I'm sitting still, whereas when Mapillary is running, it's usually in the 2 digit range, often above the 15m limit.

My best guess is that the app is devoting too much CPU priority to something other than GPS location.

Samsung Galaxy S7 Edge, Android 7.0, build number G935AUCS4BQH1

Please let me know if there is a specific scenario you'd like tested.  My time is kinda limited for the nest 10 days, but I'll do what I can.

**Comment by echeloni2p** [comment]
HI
do you use WIFI position or ONLY GPS?
echelon

**Comment by jfburke3** [comment]
@echeloni2p I've tried with Google Location Services on and off.  In many of the places I have the problem, there is no wifi within 20 miles or more.

I was able to do some additional testing and have these comments on the problem:

* During one of the "bad fix" periods, if I switch over to one of my other GPS apps (e.g., GPS Status), then it also reflects an inaccuracy in the 15-25 range, similar to what Mapillary reports.  If I *exit* Mapillary, then go over to GPS Status, the location accuracy quickly settles down to single digits.  Because of this, I think that Mapillary is causing the accuracy issues, and the problem was introduced sometime in one of the recent builds.

* If I switch to time-based capture, I can't say that the problem disappears completely, but it seems to be significantly reduced.  Except I dislike using time-based capture, because then I have to remember to stop/start sequences at every traffic light.

* I also tried running Mapillary on my Samsung Galaxy S5, Android 6.0.1, distance capture mode.  The problem still happened, although subjectively it seemed to not be as bad.

* The problem happens even after a fresh boot of either phone.

* OSC doesn't seem to have a similar problem on either phone.

My best guess about the problem is that something has been added in one of the recent builds that is causing the CPU to be too busy to get frequent enough updates from the GPS module, and so location accuracy suffers.

**Comment by jfburke3** [comment]
I seem to have two issues open on the same problem #2692

**Comment by ovivoda** [comment]
@jfburke3 Please try this version here and let me know if this one works better: 

http://android-releases.mapillary.io/Mapillary_Android_v3.102-SNAPSHOT1.apk

**Comment by jfburke3** [comment]
I decided to download Mapillary_Android_v3.103.apk, which installs fine.  But, I don't know if that build has your changes for this in it or not.....

**Comment by jfburke3** [comment]
I hate to report that 3.103 from the download website and 3.104 from the play store beta channel don't seem to have much improvement here (maybe a little, it's all subjective).

It's probably worth noting that the last several days i was shooting sequences (last weekend, Friday and Saturday) were pretty clear weather-wise, and I generally had good views of the sky. Friday was a small exception in that i did drive through an area with cloud cover for a bit, but even before and after that, i still had GPS accuracy problems. Friday was mostly Interstate driving, so i had a broad open space with little to no tree coverage to interfere with GPS reception.

**Comment by echeloni2p** [comment]
Hi
Just have to add: new Ulefon T1 arrived, it has some bad GPS receivement with A-GPS built in. But orux maps has a chance to get the position quite fine on that mobile, and even Pokemon go got a somewhat GPS position from it.
But mapillary has issues getting a gps position from it, I´ve already sent a tracelog to you.
https://www.mapillary.com/app/user/aceini?lat=47.06262883561277&lng=15.464236783847042&z=14.534676099887912&username%5B%5D=aceini&dateTo=2017-10-07&dateFrom=2017-10-07 - the outer ring was made with the Ulefon.
https://www.mapillary.com/app/user/aceini?lat=47.060858422113256&lng=15.471819462962912&z=17&username%5B%5D=aceini&dateTo=2017-10-07&dateFrom=2017-10-07&pKey=9Ez7cwULDQIz3puoFrH16Q&focus=map looks really strange...

echelon

**Comment by ovivoda** [comment]
Hi @echeloni2p @jfburke3 

We added a new setting into this build here:

http://android-releases.mapillary.io/Mapillary_Android_v3.105-SNAPSHOT1.apk

The setting allows more relaxed settings in terms of location fix accuracy. The setting name is "Relaxed accuracy limit" and you can find it in Settings->Developer. You can turn this ON when you cannot get a good location fix. Please let me know how it goes.

**Comment by jfburke3** [comment]
I've installed the snapshot build and will try that workaround setting. It would still be better to figure out what caused Mapillary to suddenly develop accuracy problems, though. 

Just to reiterate: other GPS apps don't exhibit such problems as long as Mapillary isn't running. If I exit Mapillary and run OSMTracker, it usually shows a rock-solid 5m accuracy in the same place Mapillary complains about worse-than-15m.

**Comment by polarbearing** [comment]
There is, physically, no such thing as a "rock-solid 5m accuracy". GNSS accuracies are determined by probabilities, not fixes length boundaries.

**Comment by jfburke3** [comment]
@polarbearing Let me rephrase that, then.  The other GPS app settles down at about 5m accuracy with little to no jumping around as I drive, and certainly not up to the >15m accuracy-error that Mapillary has recently begun experiencing.

**Comment by jfburke3** [comment]
Subjectively, it seems that 3.107 doesn't suffer from this problem as much.  I drove through the mountains for several hours on Sunday, and seems to have much fewer GPS accuracy warnings.  Still had some, but it seemed to be not as bad.

**Comment by jfburke3** [comment]
@ovivoda I've noticed a dramatic drop in this problem with the 3.113- 3.113-SNAPSHOT2 build you cooked up to help with #2729.

And by "dramatic drop" I mean that I'm hard-pressed to be able to cite a specific instance where I saw it occur. I still need to upload sequences and check them out, but... it's looking good so far.

**Comment by jfburke3** [comment]
As with the other issues I'm updating, I've given it a fairly good workout, and I am not seeing this problem any more with the last snapshot builds you provided, and the 3.113 beta version in the play store works pretty well, too.  I think you can close this one!

**Comment by ovivoda** [comment]
Thanks @jfburke3 Closing it!

**Comment by jfburke3** [comment]
I just had to add a comment that whatever you did to fix this, has been rock-solid for the past couple of months.  Currently on 3.119.  Fantastic performance.  A very big thank-you for finding the problem @ovivoda .
